src/common/commonutils/SshUtils.c

```diff
@@ -220,171 +235,180 @@ int CheckAppropriateCiphersForSsh(const char** ciphers, unsigned int numberOfCip
+
+        if ((NULL != expectedValue) && (0 != strcmp(value, expectedValue)))
         {
-            OsConfigLogInfo(log, "CheckLimitedUserAcccessForSsh: '%s' found in SSH Server response set to '%s'", values[i], value);
-            FREE_MEMORY(value);
-            oneFound = true;
-            break;
+            OsConfigLogError(log, "CheckSshOptionIsSet: '%s' is not set to '%s' in SSH Server response (but to '%s')", option, expectedValue, value);
+            OsConfigCaptureReason(reason, "'%s' is not set to '%s' in SSH Server response (but to '%s')",
+                "%s, also '%s' is not set to '%s' in SSH Server response (but to '%s')", option, expectedValue, value);
+            status = ENOENT;
         }
-        else
+
+        if (NULL != actualValue)
         {
-            OsConfigLogError(log, "CheckLimitedUserAcccessForSsh: '%s' not found in SSH Server response", values[i]);
-            OsConfigCaptureReason(reason, "'%s' not found in SSH Server response", "%s, also '%s' is not found in SSH server response", values[i]);
+            *actualValue = DuplicateString(value);
         }
-    }
 
-    status = oneFound ? 0 : ENOENT;
+        FREE_MEMORY(value);
+    }
+    else
+    {
+        OsConfigLogError(log, "CheckSshOptionIsSet: '%s' not found in SSH Server response", option);
+        OsConfigCaptureReason(reason, "'%s' not found in SSH Server response", "%s, also '%s' is not found in SSH server response", option);
+        status = ENOENT;
+    }
 
-    OsConfigLogInfo(log, "CheckLimitedUserAcccessForSsh: %s (%d)", status ? "failed" : "passed", status);
+    OsConfigLogInfo(log, "CheckSshOptionIsSet: %s (%d)", status ? "failed" : "passed", status);
 
     return status;
 }
 
-int CheckSshOptionIsSetToString(const char* option, const char* expectedValue, char** reason, void* log)
+static int CheckSshOptionIsSetToInteger(const char* option, int* expectedValue, int* actualValue, char** reason, void* log)
 {
-    char* value = NULL;
-    int status = 0;
+    char* actualValueString = NULL;
+    char* expectedValueString = expectedValue ? FormatAllocateString("%d", *expectedValue) : NULL;
+    int status = CheckSshOptionIsSet(option, expectedValueString, &actualValueString, reason, log);
 
-    if ((NULL == option) || (NULL == expectedValue))
+    if ((0 == status) && (NULL != actualValue))
     {
-        OsConfigLogError(log, "CheckSshOptionIsSetToString: invalid arguments (%s, %s)", option, expectedValue);
-        return EINVAL;
+        *actualValue = (NULL != actualValueString) ? atoi(actualValueString) : -1;
     }
 
-    if (false == IsSshServerActive(log))
+    FREE_MEMORY(actualValueString);
+    FREE_MEMORY(expectedValueString);
+
+    return status;
+}
+
+int CheckSshClientAliveInterval(char** reason, void* log)
+{
+    int actualValue = 0;
+    int status = CheckSshOptionIsSetToInteger("clientaliveinterval", NULL, &actualValue, reason, log);
```

Comment:
Can we turn `clientaliveinterval` into a constant?

Reply:
It's a constant. In general, where the constant is used in one place, it's coded in there. If it used in multiple places, it's defined like a real constant.